scan-translations.ts

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/env tsx
+
+/**
+ * Translation Key Scanner for Lingo.dev Integration
+ * 
+ * This script scans the web app for translation keys and validates them against
+ * the translation files. It detects missing keys and unused keys.
+ * 
+ * Usage:
+ *   pnpm scan-translations
+ * 
+ * Exit codes:
+ *   0: Success, no issues found
+ *   1: Validation errors found (missing or unused keys)
+ */
+
+import * as fs from "fs";
+import * as path from "path";
+import { glob } from "glob";
+
+// Configuration
+const WEB_APP_DIR = path.join(__dirname, "apps", "web");
+const LOCALES_DIR = path.join(WEB_APP_DIR, "locales");
+const DEFAULT_LOCALE = "en-US";
+
+// Patterns to match translation keys
+const TRANSLATION_PATTERNS = [
+    // Pattern: t("key") or t('key')
+    /\bt\s*\(\s*["']([^"']+)["']/g,
+    // Pattern: t(`key`)
+    /\bt\s*\(\s*`([^`]+)`/g,
+    // Pattern: <Trans i18nKey="key" /> or <Trans i18nKey='key' />
+    /i18nKey\s*=\s*["']([^"']+)["']/g,
+    // Pattern: <Trans i18nKey={"key"} /> or <Trans i18nKey={'key'} />
+    /i18nKey\s*=\s*\{\s*["']([^"']+)["']\s*\}/g,
+];
+
+// Directories and files to exclude from scanning
+const EXCLUDE_DIRS = [
+    "**/node_modules/**",
+    "**/dist/**",
+    "**/build/**",
+    "**/.next/**",
+    "**/coverage/**",
+    "**/locales/**",
+    "**/*.test.ts",
+    "**/*.test.tsx",
+    "**/*.spec.ts",
+    "**/*.spec.tsx",
+];
+
+interface TranslationKeys {
+    [key: string]: string | TranslationKeys;
+}
+
+interface ScanResults {
+    usedKeys: Set<string>;
+    translationKeys: Set<string>;
+    missingKeys: string[];
```

Comment:
since the keys are unique, I think the `missingKeys` and `unusedKeys` should also se sets of strings instead of array like the above two 🙏